src/service/BlockService.ts

```diff
@@ -64,6 +71,19 @@ export class BlockService implements IBlockService {
             .pipe(catchError(() => of(false)));
     }
 
+    /**
+     * Calculate transactions merkle root hash from a block
+     * @param height block height
+     * @returns root hash
+     */
+    public calculateTransactionsMerkleRootHash(height: UInt64): Observable<string> {
+        const streamer = new TransactionPaginationStreamer(this.transactionRepository);
+        return streamer
+            .search({ group: TransactionGroup.Confirmed, height: height })
+            .pipe(toArray())
```

Comment:
If possible, avoid toArray(). This puts all the transactions of the given height in memory. Can rest sort by index?

If not possible (due to the sort below), map them to ({index, hash}) before doing the to Array. Then sort that list of pair object